test/utils/PathUtils.test.ts

```diff
@@ -170,9 +170,9 @@ describe('PathUtils', () => {
       resolveSoftIndexing(rules, true);
       expect(rules.map(r => r.path)).toEqual([
         'component[Bread]',
-        'component[1]',
-        'component[1]',
-        'component[2]'
+        'component[0]',
+        'component[0]',
+        'component[1]'
```

Comment:
From your PR description:
> This PR changes the way soft index resolution works with manualSliceOrdering enabled to make it more accurately fulfill the requirement of "no access to elements with slice names without using the slice name." As a result, while the index values returned from convertSoftIndicesStrict have changed, the resulting instances are the same.

Hmmm.... As long as the author is using `[+]`, I think it makes sense to them for it to come after `[Bread]`. But if the author uses literal `[0]`, I think they would expect it to be the _first_ element in the array. I know I would.

If `[0]` _isn't_ always the first element in the array, then I think we might be violating the [Array Paths Using Numerical Indices](https://build.fhir.org/ig/HL7/fhir-shorthand/reference.html#array-paths-using-numerical-indices) part of the spec.

So, IMO, I'm not sure this is a good change. Like I said, when thinking about it in turns of soft-indexing, it's fine. But I think that what you did to make this work also affects numeric indexing (right?) - and I'm not sure that's OK.

Reply:
Unfortunately, having looked at the spec a bit more closely, I think you're right. I think I got caught up in the idea that manual slice ordering means that an array element without a slice name can't be referenced without the slice name, when that's technically not in the spec. All we can try to guarantee is that with manual slice ordering turned on, if you use soft indexing and named slices, none of your soft indices without slice names will end up referring to an element with a slice name. But, if an author makes a bunch of slices, and then a bunch of rules with hard numeric indices, we can't make that guarantee. This matches up with the descriptions in the [SUSHI docs](https://fshschool.org/docs/sushi/tips/#manual-slice-ordering).

Somehow, I got the idea in my head of a much stricter restriction: that even hard numeric indices wouldn't be able to access named slices when manual slice ordering is enabled. Then I went and implemented that. So, I'll revert that, and adjust the implementation accordingly. Thank you for reminding me of what we've already written in the FSH spec and SUSHI docs!

Reply:
Just re-read some of the SUSHI docs, and this appears in the section about manual slice ordering:

```
Instance: ExampleByName
InstanceOf: ExampleBPObservation
// some required elements omitted for brevity
* component[systolicBP].valueQuantity = 108 'mm[Hg]'
* component[diastolicBP].valueQuantity = 45 'mm[Hg]'

Instance: ExampleByNumber
InstanceOf: ExampleBPObservation
// some required elements omitted for brevity
* component[0].valueQuantity = 108 'mm[Hg]'
* component[1].valueQuantity = 45 'mm[Hg]'
```

> When manual slice ordering is enabled, rules that set values on the systolicBP or diastolicBP slices must use the slice name. With this option enabled, the ExampleByName instance would produce the same entries in component, but the ExampleByNumber instance would contain four entries in the component list: two entries that are not part of a named slice, followed by the systolicBP slice, and finally the diastolicBP slice.

This appears to directly contradict the preceding part of that section of the SUSHI docs. So, this part of the SUSHI docs may need to be revised.

Reply:
I don't think it contradicts it. Remember, if you don't explicitly refer to a required slice (with `manualSliceOrdering` on), then the required slices will go at the _end_ of the array. So `component[0]` and `component[1]` are not overwriting anything since they go first -- and then the required slices are applied _after_ that.

Reply:
It's only with `manualSliceOrdering` OFF that the required slices would get written first, and _then_ `component[0]` and `component[1]` would overwrite them.

Reply:
@cmoesel you're right: there's no contradiction here. I've clearly been down in the slice mines for too long.